### DISCUSSION TITLE
Get results from previous run

### DISCUSSION THREAD

**Discussion by hmoffatt** [CISOfy/lynis#1196] [Q&A] [answered]
Is there a way to get the nice formatted results as shown at the end of a run, from a previous run?

The log file and the report file seem to be machine rather than human readable.

**Comment by mboelen** [comment, marked as answer]
Yes, Lynis is intended to show readable output on the screen, debug information in the log, and machine-parsable information in the report file.

Now the definition of "nice" and "formatted" differs between people. So let's assume something that you can quickly see the differences between the scan of yesterday of today. What you can do is create a [diff between two Lynis scans](https://linux-audit.com/find-differences-between-two-daily-lynis-audits/):
`diff -y /var/log/lynis-report-previous.dat /var/log/lynis-report.dat | grep -v "report_datetime"`

See the link on how to create this into a script if you want to automate things.

Does this help?

**Reply by hmoffatt** [reply]
Thanks, that's a useful suggestion about comparing results.

I would find it useful to have a switch to write the readable output to file (HTML perhaps to preserve the colors and links), though I see I can redirect it to file (with --no-colors to suppress the escape sequences), which is close enough.

**Reply by mboelen** [reply]
There are tools available that can convert terminal output to HTML directly. For example the tool [terminal-to-html](https://github.com/buildkite/terminal-to-html).

As a bonus: you can then store both the raw screen output and perform a diff. And even that diff you could convert to HTML. In other words, enough possibilities.

**Reply by hmoffatt** [reply]
Thanks, I didn't know of such tools.
